core/aws_ddk_core/pipelines/state_machine_stage.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from typing import List, Optional
+
+from aws_cdk.aws_events import EventPattern, IRuleTarget
+from aws_cdk.aws_events_targets import SfnStateMachine
+from aws_cdk.aws_stepfunctions import IChainable, StateMachine, StateMachineType
+from aws_cdk.aws_stepfunctions_tasks import EventBridgePutEvents, EventBridgePutEventsEntry
+from aws_ddk_core.pipelines.stage import DataStage
+from aws_ddk_core.resources import StepFunctionsFactory
+from constructs import Construct
+
+
+class StateMachineStage(DataStage):
```

Comment:
So are we going to try building the `stages` on top of this new class `StateMachineStage()`?

Reply:
Yes, some if not all of them. I will add an example to this PR  - will refactor Glue stage to use this as the base

Reply:
Awesome! Great work Anton 😊